### ISSUE TITLE
Tofu client should send terraform-version (or tofu-version) when creating new workspace

### ISSUE THREAD

**Issue by AsafZalcman** [opentofu/opentofu#824]
### OpenTofu Version

```shell
OpenTofu v1.6.0-alpha3
on linux_arm64
```

### OpenTofu Configuration Files

terraform {
  backend "remote" {
     hostname = "<hostname to private terraform registry>"
     organization = "<organization in private terraform registry>""
     workspaces {
         name = "test"
     }
  }
}

### Debug Output

2023-11-06T10:31:22.466Z [INFO]  OpenTofu version: 1.6.0 alpha3
       2023-11-06T10:31:22.468Z [DEBUG] using github.com/hashicorp/go-tfe v1.32.1
       2023-11-06T10:31:22.468Z [DEBUG] using github.com/hashicorp/hcl/v2 v2.18.1
       2023-11-06T10:31:22.468Z [DEBUG] using github.com/hashicorp/terraform-svchost v0.1.1
       2023-11-06T10:31:22.468Z [DEBUG] using github.com/zclconf/go-cty v1.14.1
       2023-11-06T10:31:22.468Z [INFO]  Go runtime version: go1.21.3
       2023-11-06T10:31:22.468Z [INFO]  CLI args: []string{"tofu", "init"}
       2023-11-06T10:31:22.468Z [TRACE] Stdout is not a terminal
       2023-11-06T10:31:22.468Z [TRACE] Stderr is not a terminal
       2023-11-06T10:31:22.468Z [TRACE] Stdin is not a terminal
       2023-11-06T10:31:22.468Z [DEBUG] Attempting to open CLI config file: /root/.tofurc
       2023-11-06T10:31:22.468Z [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
       2023-11-06T10:31:22.468Z [INFO]  Loading CLI configuration from /root/.terraform.d/credentials.tfrc.json
       2023-11-06T10:31:22.468Z [DEBUG] ignoring non-existing provider search directory terraform.d/plugins
       2023-11-06T10:31:22.468Z [DEBUG] ignoring non-existing provider search directory /root/.terraform.d/plugins
       2023-11-06T10:31:22.468Z [DEBUG] ignoring non-existing provider search directory /root/.local/share/terraform/plugins
       2023-11-06T10:31:22.468Z [DEBUG] ignoring non-existing provider search directory /usr/local/share/terraform/plugins
       2023-11-06T10:31:22.468Z [DEBUG] ignoring non-existing provider search directory /usr/share/terraform/plugins
       2023-11-06T10:31:22.469Z [DEBUG] Found the config directory: /root/.terraform.d
       2023-11-06T10:31:22.469Z [INFO]  CLI command args: []string{"init"}

       Initializing the backend...
       2023-11-06T10:31:22.469Z [TRACE] Meta.Backend: built configuration for "remote" backend with hash value 2792484100
       2023-11-06T10:31:22.469Z [TRACE] Meta.Backend: backend has not previously been initialized in this working directory
       2023-11-06T10:31:22.469Z [DEBUG] New state was assigned lineage "5f70810e-357c-c542-c2e6-3b3d0f85ce52"
       2023-11-06T10:31:22.469Z [TRACE] Meta.Backend: moving from default local state only to "remote" backend
       2023-11-06T10:31:22.469Z [DEBUG] checking for provisioner in "."
       2023-11-06T10:31:22.470Z [DEBUG] checking for provisioner in "/usr/local/bin"
       2023-11-06T10:31:22.470Z [TRACE] backend/local: state manager for workspace "default" will:
        - read initial snapshot from terraform.tfstate
        - write new snapshots to terraform.tfstate
        - create any backup at terraform.tfstate.backup
       2023-11-06T10:31:22.470Z [TRACE] statemgr.Filesystem: reading initial snapshot from terraform.tfstate
       2023-11-06T10:31:22.470Z [TRACE] statemgr.Filesystem: snapshot file has nil snapshot, but that's okay
       2023-11-06T10:31:22.470Z [TRACE] statemgr.Filesystem: read nil snapshot
       2023-11-06T10:31:22.470Z [TRACE] Meta.Backend: ignoring local "default" workspace because its state is empty
       2023-11-06T10:31:22.470Z [DEBUG] Service discovery for tflocal.com at https://tflocal.com/.well-known/terraform.json
       2023-11-06T10:31:22.470Z [TRACE] HTTP client GET request to https://tflocal.com/.well-known/terraform.json
       2023-11-06T10:31:22.547Z [TRACE] Meta.BackendForPlan: backend *remote.Remote supports operations
       2023-11-06T10:31:22.548Z [DEBUG] Service discovery for tflocal.com aliased as localterraform.com
       2023-11-06T10:31:22.548Z [DEBUG] New state was assigned lineage "586927fe-ceb3-e127-3d15-d6fcfc239a2b"
       2023-11-06T10:31:22.562Z [TRACE] Preserving existing state lineage "586927fe-ceb3-e127-3d15-d6fcfc239a2b"

      
       Successfully configured the backend "remote"! OpenTofu will automatically
       use this backend unless the backend configuration changes.
        Error: No existing workspaces.
     
      Use the "tofu workspace" command to create and select a new workspace.
       If the backend already contains existing workspaces, you may need to update
      the backend configuration.
      
       2023-11-06T10:31:22.581Z [INFO]  OpenTofu version: 1.6.0 alpha3
       2023-11-06T10:31:22.581Z [DEBUG] using github.com/hashicorp/go-tfe v1.32.1
       2023-11-06T10:31:22.581Z [DEBUG] using github.com/hashicorp/hcl/v2 v2.18.1
       2023-11-06T10:31:22.581Z [DEBUG] using github.com/hashicorp/terraform-svchost v0.1.1
       2023-11-06T10:31:22.581Z [DEBUG] using github.com/zclconf/go-cty v1.14.1
       2023-11-06T10:31:22.581Z [INFO]  Go runtime version: go1.21.3
       2023-11-06T10:31:22.581Z [INFO]  CLI args: []string{"tofu", "workspace", "new", "test1"}
       2023-11-06T10:31:22.581Z [TRACE] Stdout is not a terminal
       2023-11-06T10:31:22.581Z [TRACE] Stderr is not a terminal
       2023-11-06T10:31:22.581Z [TRACE] Stdin is not a terminal
       2023-11-06T10:31:22.581Z [DEBUG] Attempting to open CLI config file: /root/.tofurc
       2023-11-06T10:31:22.581Z [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
       2023-11-06T10:31:22.581Z [INFO]  Loading CLI configuration from /root/.terraform.d/credentials.tfrc.json
       2023-11-06T10:31:22.581Z [DEBUG] ignoring non-existing provider search directory terraform.d/plugins
       2023-11-06T10:31:22.581Z [DEBUG] ignoring non-existing provider search directory /root/.terraform.d/plugins
       2023-11-06T10:31:22.581Z [DEBUG] ignoring non-existing provider search directory /root/.local/share/terraform/plugins
       2023-11-06T10:31:22.581Z [DEBUG] ignoring non-existing provider search directory /usr/local/share/terraform/plugins
       2023-11-06T10:31:22.581Z [DEBUG] ignoring non-existing provider search directory /usr/share/terraform/plugins
       2023-11-06T10:31:22.581Z [DEBUG] Found the config directory: /root/.terraform.d
       2023-11-06T10:31:22.581Z [INFO]  CLI command args: []string{"workspace", "new", "test1"}
       2023-11-06T10:31:22.582Z [TRACE] Meta.Backend: built configuration for "remote" backend with hash value 2792484100
       2023-11-06T10:31:22.582Z [TRACE] Preserving existing state lineage "586927fe-ceb3-e127-3d15-d6fcfc239a2b"
       2023-11-06T10:31:22.582Z [TRACE] Preserving existing state lineage "586927fe-ceb3-e127-3d15-d6fcfc239a2b"
       2023-11-06T10:31:22.582Z [TRACE] Meta.Backend: working directory was previously initialized for "remote" backend
       2023-11-06T10:31:22.582Z [TRACE] Meta.Backend: using already-initialized, unchanged "remote" backend configuration
       2023-11-06T10:31:22.582Z [DEBUG] Service discovery for tflocal.com at https://tflocal.com/.well-known/terraform.json
       2023-11-06T10:31:22.582Z [TRACE] HTTP client GET request to https://tflocal.com/.well-known/terraform.json
       2023-11-06T10:31:22.618Z [TRACE] Meta.BackendForPlan: backend *remote.Remote supports operations
       2023-11-06T10:31:22.618Z [DEBUG] Service discovery for tflocal.com aliased as localterraform.com
       2023-11-06T10:31:22.618Z [TRACE] Meta.Backend: instantiated backend of type *remote.Remote
       2023-11-06T10:31:22.618Z [DEBUG] checking for provisioner in "."
       2023-11-06T10:31:22.618Z [DEBUG] checking for provisioner in "/usr/local/bin"
       2023-11-06T10:31:22.618Z [TRACE] Meta.Backend: backend *remote.Remote supports operations
       Remote workspace OpenTofu version "" does not match local OpenTofu version "1.6.0-alpha3"

### Expected Behavior

tofu should be able to create a workspace

### Actual Behavior

Tofu failing to create a workspace using tofu workspace new command

### Steps to Reproduce

tofu init
tofu workspace new test

**Comment by Yantrio** [comment]
@AsafZalcman Would you be able to tell us what you are using as the remote http backend to help us reproduce this issue?

**Comment by AsafZalcman** [comment]
I am using jfrog artifactory instance - which supports TF currently

**Comment by AsafZalcman** [comment]
The problem is in the payload of

POST remote/v2/organizations/<orgName>/workspaces
see https://developer.hashicorp.com/terraform/cloud-docs/api-docs/workspaces#create-a-workspace

/artifactory/api/terraform/remote/v2/organizations/terraformbackend-local-z9y6oftjho/workspaces
--

**Comment by Yantrio** [comment]
Thanks for reporting this! I've identified the issue, but I'm not sure on the best way to approach this going forward.

https://github.com/opentofu/opentofu/blob/487d9bc6a488b236e3cac8a9473568118736ef7f/internal/backend/remote/backend.go#L666-L670

Right now, we don't attach the version if the `tofu` version if we're in a pre-release or RC. 

Until our stable release has happened this will be causing issues.

We have a couple of options though, we could remove this check and always store the version, but this may break some functionality with existing tooling that consumes this data.

I'll ask other members of the core team to chime in here and see what they say.

@kislerdm @cube2222 @RLRabinowitz @cam72cam What do you guys think?

**Comment by kislerdm** [comment]
@Yantrio Hey James! Thank you for the link to the code. 

WDYT about setting the flag [`ignoreVersionConflict`](https://github.com/opentofu/opentofu/blob/487d9bc6a488b236e3cac8a9473568118736ef7f/internal/backend/remote/backend.go#L683C16-L683C16) until the stable release?

**Comment by cube2222** [comment]
@Yantrio My **opinion** is that as far as state backends go, we should always post the current version **with the prerelease suffix removed**. So 1.6.x-abc becomes 1.6.x. And that should work for any kind of prerelease (including alpha).

**Comment by Yantrio** [comment]
I think overall that adding the `ignoreVersionConflict` flag here is just papering over the issue at hand.

In my opinion, this check isn't really needed as it was added way back in 0.x terraform days and is intended behavior for TFE. 

I would like to propose that we go with what @cube2222 has pitched and do the following:
- Remove the check, and always attach the terraform version to the workspace creation options
- Trim the version so that we only attach `major.minor.patch` and not rc/prerelease information

We should let people choose if they want to use an alpha/pre-release build with http backends and not lock them into a choice by having this check there.

**Comment by kislerdm** [comment]
Hey folks! I've done a research to compare behaviour of tofu vs terraform when interacting with JFrog in case of absence of remote workspace. Moreover, I used TFE as a reference. Note that TFE was used with terraform only.

**BLUF**: both, opentofu (_alpha3_ was tested) and terraform (_beta3_ was tested), use empty string when attempting to create new remote workspace during the init step. However, terraform uses `ignoreVersionConflict` attribute set to `true` by default. It leads for terraform to complete the init step successfully.

## Open questions

- Shall we replicate the terraform's behaviour and set `ignoreVersionConflict` to `true` by default?
- Shall we account for the case when a remote JFrog workspace was created using terraform pre-release? For context, our proposed [fix](https://github.com/opentofu/opentofu/issues/824#issuecomment-1795311429) does not cover this case. The conclusion is based on the following assumption: if a user wold create a JFrog workspace with a terraform pre-released version, and use tofu afterwards, the init step would fail because of the [check](https://github.com/opentofu/opentofu/blob/487d9bc6a488b236e3cac8a9473568118736ef7f/internal/backend/remote/backend.go#L683-L690).

## Details

**Note** that the terraform version was selected arbitrary following a single criterium only: it had to be pre-release (not even a candidate, i.e. an "-rc" was beyond consideration).

### JFrog: tofu vs. terraform init step

**GIVEN**

- No pre-defined remote workspace existed
- The following backend config:

```hcl
terraform {
  backend "remote" {
    hostname     = "{{YouJFrogOrgName}}.jfrog.io"
    organization = "{{YouLocalTFRepoName}}"
    workspaces {
      name = "foo"
    }
  }
}
```
-  Versions:

|OpenTofu|Terraform|
|:---|:---|
|v1.6.0-alpha3|v1.6.0-beta3|

**WHEN**

Run the init step:

|OpenTofu|Terraform|
|:---|:---|
|`toru init`|`terraform init`|

**THEN**

- Execution status:

|OpenTofu|Terraform|
|:---|:---|
|Fails|Succeeds|

- The workspace is created in the remote backend

<details>
<summary><h4>JFrog remote workspace meta</h4></summary>

```json
{
  "data": {
    "attributes": {
      "actions": null,
      "description": null,
      "environment": null,
      "locked": false,
      "name": "foo",
      "operations": false,
      "permissions": null,
      "source": null,
      "allow-destroy-plan": false,
      "apply-duration-average": null,
      "auto-apply": false,
      "auto-destroy-at": null,
      "created-at": null,
      "execution-mode": null,
      "file-triggers-enabled": false,
      "global-remote-state": false,
      "latest-change-at": null,
      "plan-duration-average": null,
      "policy-check-failures": null,
      "queue-all-runs": false,
      "resource-count": 0,
      "run-failures": null,
      "source-name": null,
      "source-url": null,
      "speculative-enabled": false,
      "structured-run-output-enabled": false,
      "terraform-version": null,
      "trigger-prefixes": null,
      "updated-at": null,
      "vcs-repo": null,
      "vcs-repo-identifier": null,
      "working-directory": null,
      "workspace-kpis-runs-count": null
    },
    "id": "backend__foo",
    "links": null,
    "relationships": null,
    "type": "workspaces"
  }
}
```
</details>

### Logs


|OpenTofu|Terraform|Comment|
|:---|:---|:---|
|`internal/backend/remote/backend.go:645: open-tofu-824b.StateMgr: b = &remote.Remote{CLI:(*cli.ConcurrentUi)(0x140001199e0), CLIColor:(*colorstring.Colorize)(0x1400085f100), ContextOpts:(*tofu.ContextOpts)(0x14000990d70), client:(*tfe.Client)(0x14000810480), lastRetry:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}, hostname:"{{YouJFrogOrgName}}.jfrog.io", organization:"{{YouLocalTFRepoName}}", workspace:"foo", prefix:"", services:(*disco.Disco)(0x140006fdb80), local:(*local.Local)(0x140000c4b40), forceLocal:true, opLock:sync.Mutex{state:0, sema:0x0}, ignoreVersionConflict:false}`|`internal/backend/remote/backend.go:640: open-tofu-824b.StateMgr: b = &remote.Remote{CLI:(*cli.ConcurrentUi)(0x14000880be8), CLIColor:(*colorstring.Colorize)(0x1400092e820), ContextOpts:(*terraform.ContextOpts)(0x14000664d20), client:(*tfe.Client)(0x140006f1680), lastRetry:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}, hostname:"{{YouJFrogOrgName}}.jfrog.io", organization:"{{YouLocalTFRepoName}}", workspace:"foo", prefix:"", services:(*disco.Disco)(0x14000862b40), local:(*local.Local)(0x14000858be0), forceLocal:true, opLock:sync.Mutex{state:0, sema:0x0}, ignoreVersionConflict:true}`|It can be noticed the the `ignoreVersionConflict` values differ: `true` for terraform, `false` for tofu.|
|`internal/backend/remote/backend.go:668: open-tofu-824workspace foo not found, attempt to create`|`internal/backend/remote/backend.go:663: open-tofu-824workspace foo not found, attempt to create`||
|`internal/backend/remote/backend.go:680: open-tofu-824WorkspaceCreateOptions = tfe.WorkspaceCreateOptions{Type:"", AgentPoolID:(*string)(nil), AllowDestroyPlan:(*bool)(nil), AssessmentsEnabled:(*bool)(nil), AutoApply:(*bool)(nil), Description:(*string)(nil), ExecutionMode:(*string)(nil), FileTriggersEnabled:(*bool)(nil), GlobalRemoteState:(*bool)(nil), MigrationEnvironment:(*string)(nil), Name:(*string)(0x140007a63c0), Operations:(*bool)(nil), QueueAllRuns:(*bool)(nil), SpeculativeEnabled:(*bool)(nil), SourceName:(*string)(nil), SourceURL:(*string)(nil), StructuredRunOutputEnabled:(*bool)(nil), TerraformVersion:(*string)(nil), TriggerPrefixes:[]string(nil), TriggerPatterns:[]string(nil), VCSRepo:(*tfe.VCSRepoOptions)(nil), WorkingDirectory:(*string)(nil), Tags:[]*tfe.Tag(nil), Project:(*tfe.Project)(nil)}`|`internal/backend/remote/backend.go:675: open-tofu-824WorkspaceCreateOptions = tfe.WorkspaceCreateOptions{Type:"", AgentPoolID:(*string)(nil), AllowDestroyPlan:(*bool)(nil), AssessmentsEnabled:(*bool)(nil), AutoApply:(*bool)(nil), Description:(*string)(nil), ExecutionMode:(*string)(nil), FileTriggersEnabled:(*bool)(nil), GlobalRemoteState:(*bool)(nil), MigrationEnvironment:(*string)(nil), Name:(*string)(0x1400092ec90), Operations:(*bool)(nil), QueueAllRuns:(*bool)(nil), SpeculativeEnabled:(*bool)(nil), SourceName:(*string)(nil), SourceURL:(*string)(nil), StructuredRunOutputEnabled:(*bool)(nil), TerraformVersion:(*string)(nil), TriggerPrefixes:[]string(nil), TriggerPatterns:[]string(nil), VCSRepo:(*tfe.VCSRepoOptions)(nil), WorkingDirectory:(*string)(nil), Tags:[]*tfe.Tag(nil), Project:(*tfe.Project)(nil)}`|It can be noticed that the `WorkspaceCreateOptions` values are consistent. In particular, the value `TerraformVersion` is not set in both cases.|
|`internal/backend/remote/backend.go:688: open-tofu-824created workspace version = `|`internal/backend/remote/backend.go:683: open-tofu-824created workspace version = `|Empty string is return from the backed as the terraform version.|
|`internal/backend/remote/backend.go:698: open-tofu-824validate remote version  vs. local version 1.6.0-alpha3`||Terraform does not [check](https://github.com/opentofu/opentofu/blob/487d9bc6a488b236e3cac8a9473568118736ef7f/internal/backend/remote/backend.go#L683-L690) the versions.|


## TFE

**GIVEN**

- No pre-defined remote workspace existed
- The following backend config:

```hcl
terraform {
  backend "remote" {
    organization = "{{YourTFEOrgName}}"
    workspaces {
      name = "foo"
    }
  }
}
```
- terraform v1.6.0-beta3

**WHEN**

Run `terraform init`

**THEN**

- Execution succeeds
- The remote workspace is created with the version set to "latest" implicitly:

<img width="595" alt="image" src="https://github.com/opentofu/opentofu/assets/13434797/4bd9c9e1-f77f-4f0d-9ecc-455c0cef49d4">

## Logs

The logs show comparison of execution when using different backends.

|TFE|JFrog|Comment|
|:---|:---|:---|
|`internal/backend/remote/backend.go:640: open-tofu-824b.StateMgr: b = &remote.Remote{CLI:(*cli.ConcurrentUi)(0x1400090abe8), CLIColor:(*colorstring.Colorize)(0x140009d4840), ContextOpts:(*terraform.ContextOpts)(0x140005d7c70), client:(*tfe.Client)(0x14000935680), lastRetry:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}, hostname:"app.terraform.io", organization:"{{YourTFEOrgName}}", workspace:"foo", prefix:"", services:(*disco.Disco)(0x14000796b00), local:(*local.Local)(0x14000920dc0), forceLocal:false, opLock:sync.Mutex{state:0, sema:0x0}, ignoreVersionConflict:true}`|`internal/backend/remote/backend.go:640: open-tofu-824b.StateMgr: b = &remote.Remote{CLI:(*cli.ConcurrentUi)(0x14000880be8), CLIColor:(*colorstring.Colorize)(0x1400092e820), ContextOpts:(*terraform.ContextOpts)(0x14000664d20), client:(*tfe.Client)(0x140006f1680), lastRetry:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}, hostname:"{{YouJFrogOrgName}}.jfrog.io", organization:"{{YouLocalTFRepoName}}", workspace:"foo", prefix:"", services:(*disco.Disco)(0x14000862b40), local:(*local.Local)(0x14000858be0), forceLocal:true, opLock:sync.Mutex{state:0, sema:0x0}, ignoreVersionConflict:true}`|It can be noticed the the `forceLocal` values differ: `true` for JFrog, `false` for TFE.|
|`internal/backend/remote/backend.go:668: open-tofu-824workspace foo not found, attempt to create`|`internal/backend/remote/backend.go:663: open-tofu-824workspace foo not found, attempt to create`||
|`internal/backend/remote/backend.go:675: open-tofu-824WorkspaceCreateOptions = tfe.WorkspaceCreateOptions{Type:"", AgentPoolID:(*string)(nil), AllowDestroyPlan:(*bool)(nil), AssessmentsEnabled:(*bool)(nil), AutoApply:(*bool)(nil), Description:(*string)(nil), ExecutionMode:(*string)(nil), FileTriggersEnabled:(*bool)(nil), GlobalRemoteState:(*bool)(nil), MigrationEnvironment:(*string)(nil), Name:(*string)(0x140009d48c0), Operations:(*bool)(nil), QueueAllRuns:(*bool)(nil), SpeculativeEnabled:(*bool)(nil), SourceName:(*string)(nil), SourceURL:(*string)(nil), StructuredRunOutputEnabled:(*bool)(nil), TerraformVersion:(*string)(nil), TriggerPrefixes:[]string(nil), TriggerPatterns:[]string(nil), VCSRepo:(*tfe.VCSRepoOptions)(nil), WorkingDirectory:(*string)(nil), Tags:[]*tfe.Tag(nil), Project:(*tfe.Project)(nil)}`|`internal/backend/remote/backend.go:675: open-tofu-824WorkspaceCreateOptions = tfe.WorkspaceCreateOptions{Type:"", AgentPoolID:(*string)(nil), AllowDestroyPlan:(*bool)(nil), AssessmentsEnabled:(*bool)(nil), AutoApply:(*bool)(nil), Description:(*string)(nil), ExecutionMode:(*string)(nil), FileTriggersEnabled:(*bool)(nil), GlobalRemoteState:(*bool)(nil), MigrationEnvironment:(*string)(nil), Name:(*string)(0x1400092ec90), Operations:(*bool)(nil), QueueAllRuns:(*bool)(nil), SpeculativeEnabled:(*bool)(nil), SourceName:(*string)(nil), SourceURL:(*string)(nil), StructuredRunOutputEnabled:(*bool)(nil), TerraformVersion:(*string)(nil), TriggerPrefixes:[]string(nil), TriggerPatterns:[]string(nil), VCSRepo:(*tfe.VCSRepoOptions)(nil), WorkingDirectory:(*string)(nil), Tags:[]*tfe.Tag(nil), Project:(*tfe.Project)(nil)}`|It can be noticed that the `WorkspaceCreateOptions` values are consistent. In particular, the value `TerraformVersion` is not set in both cases.|
|`internal/backend/remote/backend.go:683: open-tofu-824created workspace version = 1.6.3`|`internal/backend/remote/backend.go:683: open-tofu-824created workspace version = `|It can be noticed that TFE sets the version implicitly and replies it back while JFrog does not make any implicit assumptions on user's behalf|

**Comment by kislerdm** [comment]
I double checked with tofu installed using homebrew on MacOS (Arch: arm64). The `tofu init` step works as expected. In other words, I could not reproduce the issue using the latest version of tofu installed from brew.

Just in case, here's the log trace with dependencies list:

```commandline
2023-11-07T17:29:50.469+0100 [INFO]  OpenTofu version: 1.6.0 dev
2023-11-07T17:29:50.470+0100 [DEBUG] using github.com/hashicorp/go-tfe v1.32.1
2023-11-07T17:29:50.470+0100 [DEBUG] using github.com/hashicorp/hcl/v2 v2.18.1
2023-11-07T17:29:50.470+0100 [DEBUG] using github.com/hashicorp/terraform-svchost v0.1.1
2023-11-07T17:29:50.470+0100 [DEBUG] using github.com/zclconf/go-cty v1.14.1
2023-11-07T17:29:50.470+0100 [INFO]  Go runtime version: go1.21.3
2023-11-07T17:29:50.470+0100 [INFO]  CLI args: []string{"tofu", "init"}
```

Here is the logs for the tofu I installed by downloading a zip archive from the github releases artefacts:

```commandline
2023-11-07T17:32:17.471+0100 [INFO]  OpenTofu version: 1.6.0 alpha3
2023-11-07T17:32:17.471+0100 [DEBUG] using github.com/hashicorp/go-tfe v1.32.1
2023-11-07T17:32:17.471+0100 [DEBUG] using github.com/hashicorp/hcl/v2 v2.18.1
2023-11-07T17:32:17.471+0100 [DEBUG] using github.com/hashicorp/terraform-svchost v0.1.1
2023-11-07T17:32:17.471+0100 [DEBUG] using github.com/zclconf/go-cty v1.14.1
2023-11-07T17:32:17.471+0100 [INFO]  Go runtime version: go1.21.3
2023-11-07T17:32:17.471+0100 [INFO]  CLI args: []string{"./tofu", "init"}
````

However, if I compile from the source using the codebase in the [state](https://github.com/opentofu/opentofu/tree/60a3f72d43c002f83246e657a15b9b2157fd74c6), the init step fails. See the [comment](https://github.com/opentofu/opentofu/issues/824#issuecomment-1798356428) for details.

Also, if I try to create, or list workspaces, I've got the error:

```commandline
workspaces not supported
```

**Comment by cam72cam** [comment]
So that implies that tofu changed the default for ignoreVersionConflict since 1.6.0 dev, have we identified when/why that took place?

**Comment by kislerdm** [comment]
@AsafZalcman Hi Asaf! I'd like to share that the issue has been resolved by the change #835.

Moreover, we decided not to change the tofu behaviour following your suggestion until the stable release: similar to terraform, tofu will not send pre-released versions to the remote backend. The rationale: we’d like to avoid implementing deviations from terraform which don’t block first stable tofu release. 

Thanks for understanding.
